pghoard/common.py

```diff
@@ -353,5 +354,25 @@ def increase_pipe_capacity(*pipes):
                 pass
 
 
+class PGHoardThread(Thread):
+    def __init__(self):
+        super().__init__()
+        self.exception: Optional[Exception] = None
+
+    def run_safe(self):
+        raise NotImplementedError()
+
+    def run(self):
+        try:
+            self.run_safe()
+        except Exception as ex:  # pylint: disable=broad-except
+            self.exception = ex
```

Comment:
I would add some kind of stats call here to be aware when it happens

Reply:
I would say that's what logs are for. Those cases are not supposed to happen, it's for when they do, at least the error pops correctly, gets logged and crashes the program instead of being silently ignored.

The cases which might happen (and that includes sometime catch-all exception, see pghoard/compressor.py) handles the stat themselves.

Reply:
Fair enough